isa.c

```diff
@@ -293,7 +298,89 @@ static void exec_B(uint32_t instruction)
         next_state.regs[LR] = curr_state.regs[PC] + 4;
         // address of instruction after the branch instruction stored in LR
     }
-    next_state.regs[PC] = curr_state.regs[PC] + (sign_extend(signed_immed_24,30,24) << 2);
+    next_state.regs[PC] = curr_state.regs[PC] + (int32_t)(sign_extend(signed_immed_24, 24, 30) << 2);
 }
 
+static void exec_BL(uint32_t instruction)
```

Comment:
@parthmittal rename this to B, as B and BL are same from instruction pov, as L bit is passed
and remove @divyanshu-talwar 's B as it has bug, he made immed_24 and 8 bit variable

also remove condition_check as it is redundant, we check it above in decoding function